spec/acceptance/should_create_task_spec.rb

```diff
@@ -67,4 +75,190 @@
     query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
     on(host, query_cmd)
   end
+
+  it "Should create a task with a username and password" do
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure        => present,
+      compatibility => 1,
+      command       => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments     => "foo bar baz",
+      working_dir   => 'c:\\\\windows',
+      user          => '#{@username}',
+      password      => '#{@password}',
+      trigger       => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider      => 'taskscheduler_api2'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
+
+    # Ensure it's idempotent
+    execute_manifest(pp, :catch_changes  => true)
+
+    # Verify the task exists
+    query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
+  end
+
+  it "Should create a task with a username and password" do
```

Comment:
"using legacy provider" ... ?

Reply:
Since we felt it necessary to test it above at line 66 I figured we should test it here. Are we worried about accidentally breaking the vendored legacy code in the future?